rclcpp/test/rclcpp/test_executor.cpp

```diff
@@ -488,6 +517,37 @@ TEST_F(TestExecutor, spin_until_future_complete_future_already_complete) {
     dummy.spin_until_future_complete(future, std::chrono::milliseconds(1)));
 }
 
+TEST_F(TestExecutor, spin_until_complete_condition_already_complete) {
+  DummyExecutor dummy;
+  auto condition = []() {return true;};
+  EXPECT_EQ(
+    rclcpp::FutureReturnCode::SUCCESS,
+    dummy.spin_until_complete(condition, std::chrono::milliseconds(1)));
+}
+
+TEST_F(TestExecutor, spin_until_complete_returns_after_condition) {
+  DummyExecutor dummy;
+  auto node = std::make_shared<rclcpp::Node>("node", "ns");
+  bool spin_called = false;
+  auto timer =
+    node->create_wall_timer(
+    std::chrono::milliseconds(1), [&]() {
+      spin_called = true;
+    });
+  dummy.add_node(node);
+  // Check that we stop spinning after the condition is ready
+  const auto condition_delay = std::chrono::milliseconds(10);
+  const auto start = std::chrono::steady_clock::now();
+  auto condition = [&]() {return std::chrono::steady_clock::now() - start > condition_delay;};
+  EXPECT_EQ(
+    rclcpp::FutureReturnCode::SUCCESS,
+    dummy.spin_until_complete(condition, std::chrono::seconds(1)));
+  EXPECT_TRUE(spin_called);
+  const auto end_delay = std::chrono::steady_clock::now() - start;
+  EXPECT_GE(end_delay, condition_delay);
+  EXPECT_LT(end_delay, std::chrono::milliseconds(500));
```

Comment:
This check doesn't look great to me, we make no guarantee that functions like `spin_until_complete()` start-up and shutdown within 500ms, discounting the actual blocking/executing time. I would just not check this. The test is still useful without it because it's checking that the condition eventually gets checked for true even though it doesn't directly wake the wait set.

Reply:
I mean they should be that fast, but also this isn't a performance test, and on a very slow computer or on a non-real-time OS, this kind of thing cannot be assumed and avoid flakes in rare cases. Increasing the time from 500ms will only ever asymptotically reduce the chances of a flake, but not avoid them completely.

Reply:
Totally agree. I removed this line. I wasn't sure if you also wanted me to remove the `EXPECT_GE(end_delay, condition_delay);` line above.

Reply:
I think that one is reasonable, since the call needs to take longer than the condition's sleep, but also I wouldn't expect it to fail. I think it's fine as-is now that you removed the last check.